### ISSUE TITLE
Lot No control in Receipt not working

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh#4012]
### Is this a bug or feature request?
Bug
### What is the current behavior?
When Banning a Lot No and doing a material receipt, then the DD Orderline created for that lot does not have a movement to the quarantine Warehouse.
#### Which are the steps to reproduce?
Open, try and see.
### What is the expected or desired behavior?
Shall work.

**Comment by metas-rc** [comment]
More findings: 

* When the asi is missing from the inout line it is also missing from the orderline.
* I did some debugging and found out the orderline is created with ASI only if the attribute set from the product category contains at least one of the attributes that have special validators.
* These attribute validators are implementing org.adempiere.mm.attributes.api.IModelAttributeSetInstanceListener
* THe cncrete classes for them are: 

![image](https://user-images.githubusercontent.com/16001442/40552306-f6fa6346-6047-11e8-9cd7-38db6cea18a3.png)

So, in conclusion, we should either add such a listener for all the attributes that we want to propagate in the inout line or to change the attribute logic.

**Comment by metas-rc** [comment]
Talked with Mark and decided on the following solution: 
* Implement a listener for the LotNumber attribute, too, like the ones for Marke ADR, Country etc.
* If the Lotnumber attribute is set in the AttributeSet , the orderline shall be created directly with an ASI => the inoutline will also have an asi in the end
* This way, the lot number will always be propagated if it was set in the HU.
* If the lotnumber does not exist in the attribute set of the product, it shall not be displayed in the HU attributes in the hu editor!

**Comment by metas-rc** [comment]
Notes for IT:

* if you want to see an attribute in the handling units editor, you must set the IsOnlyIfInAttributeSet flag in M_HU_PI_Attribute ( window Packing instructions or Packing Instructions version, tab Attributes) on false
* It is on false by default
* If the flag is on true, the attribute must be present in the product's attribute set.
*

**Comment by metas-lc** [comment]
IT

I. old case: (IsOnlyIfInAttributeSet all on N). Usual Attributes
1. Have a product with a category with attribute set without lotNo, but containing the usual ones like label, country etc
2. create a purchase order for it, receive HUs
3. set lotNo and other attributes, create receipt
=> lotno and the other attributes are set OK
.

II. new case: attribute set only with LotNo (IsOnlyIfInAttributeSet all on N)
1. Have a product with a category with attribute set ONLY with lotNo
2. create a purchase order for it, receive HUs
3. set lotNo and other attributes, create receipt
=> lotno and the other attributes are set OK
.

III. old case: (IsOnlyIfInAttributeSet all on N). Unusual Attributes
1. Have a product with a category with attribute set, but containing some attributes, except the usual ones (you can use e.g. parzelle)
2. create a purchase order for it, receive HUs
3. set lotNo and other attributes, create receipt
=> lotno and the other attributes are missing OK
.

IV. new case: IsOnlyIfInAttributeSet on Y (Packing Item Template)
1. Have a product with a category with attribute set, containing some attributes, like: label, country, marke, lotno 
2. go to Packing Instruction Version window, select Packing Item Template and set IsOnlyIfInAttributeSet on Y to country, marke, qualitydiscount, qualitynotice
3. create a purchase order for it, receive HUs
=> label, country, marke, lotno are shown OK
=> qualitydiscount, qualitynotice are hidden OK
=> rest of the attributes are shown OK
4. set lotNo and other attributes, create receipt
=> lotno and the other attributes are set OK
.

V. new case: IsOnlyIfInAttributeSet on Y (LU)
1. Have a product with a category without attribute set
2. go to Packing Instruction Version window, select the LU you usually use and set IsOnlyIfInAttributeSet on Y to e.g. label
3. create a purchase order for your product, receive HUs
=> label is hidden on LU OK
=> label is not hidden on TU OK
.

VI. old case: (IsOnlyIfInAttributeSet all on N). No Attribute set
1. Have a product with a category without attribute set 
2. create a purchase order for it, receive HUs
3. set lotNo and other attributes, create receipt
=> lotno and the other attributes are not set OK

note: i had to set IsOnlyIfInAttributeSet in swing, because i didn't have permission in webui. Created task:
https://github.com/metasfresh/metasfresh-webui-api/issues/981
